test/hotspot/jtreg/compiler/c2/TestMergeLoads.java

```diff
@@ -0,0 +1,706 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2025, Alibaba Group Holding Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.c2;
+
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Utils;
+import jdk.internal.misc.Unsafe;
+import java.lang.reflect.Array;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Random;
+
+/*
+ * @test
+ * @summary Test merging of consecutive loads
+ * @modules java.base/jdk.internal.misc
+ * @library /test/lib /
+ *
+ * @run main compiler.c2.TestMergeLoads unaligned
+ * @run main compiler.c2.TestMergeLoads aligned
+ *
+ * @requires os.arch != "riscv64" | vm.cpu.features ~= ".*zbb.*"
+ */
+
+public class TestMergeLoads {
+    static int RANGE = 1000;
+    private static final Unsafe UNSAFE = Unsafe.getUnsafe();
+    private static final Random RANDOM = Utils.getRandomInstance();
+
+    // Inputs
+    byte[] aB = new byte[RANGE];
+    char[] aC = new char[RANGE];
+    short[] aS = new short[RANGE];
+
+    interface TestFunction {
+        Object[] run(boolean isWarmUp, int rnd);
+    }
+
+    Map<String, Map<String, TestFunction>> testGroups = new HashMap<String, Map<String, TestFunction>>();
+
+    public static void main(String[] args) {
+        TestFramework framework = new TestFramework(TestMergeLoads.class);
+        framework.addFlags("--add-modules", "java.base", "--add-exports", "java.base/jdk.internal.misc=ALL-UNNAMED");
+
+        switch (args[0]) {
+            case "aligned"     -> { framework.addFlags("-XX:-UseUnalignedAccesses"); }
+            case "unaligned"   -> { framework.addFlags("-XX:+UseUnalignedAccesses"); }
```

Comment:
Can you please also add an explicit run with `StressIGVN`? Because the flag is not whitelisted for the TestFramework, and so if it was set from the outside, the IR rules would not be executed. But it would be nice that your algorithm is stable to reorderings in IGVN ;)

Reply:
I think my optimization is not dependent on order of IGVN. I will verify it with this option. Thanks.

Reply:
`StreassIGVN` tests added